Hymns/Display/HymnLyricsView.swift

```diff
@@ -21,18 +28,23 @@ public struct HymnLyricsView: View {
         return AnyView(
             ScrollView {
                 VStack(alignment: .leading) {
-                    Group {
-                        ForEach(lyrics, id: \.self) { verse in
+                    ForEach(lyrics, id: \.self) { verse in
+                        VStack { //needed to contain verticle spacer
                             Group {
-                                ForEach(verse.verseContent, id: \.self) { line in
-                                    Text(line)
-                                }
-                                Spacer().frame(height: 30)
+                                HStack(alignment: .top) { //needed for verse numbers to be displayed to the left
```

Comment:
Instead of creating an HStack to contain the verse number, is it possible just to create a separate view for line number and just align it with the verse content?

Reply:
Not easy, I tried a solution like that. My main problem is in the actual getting line numbers based off the hymn and how to put in a blank when the stanza is a chorus. I am wondering if there is a solution where we dynamically create an array based on the stanza types. For example if the chorus is the second stanza it would look something like [1, " ", 2, 3, 4]. Then during the ForEach put them in. But I still haven't found a viable solution. It's very difficult because state can not change during the ForEach loop so you have to bring you data ready to be pasted in instead of creating during the loop.

Reply:
I just pushed a VerseView.swift onto this branch. Do you think something like that could work?